pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -47,38 +70,54 @@ let
         ${replaced} = v;
         "tree-sitter-${replaced}" = v;
       }
-    ) generatedDerivations;
+    ) parsersWithMeta;
 
-  allGrammars = lib.attrValues generatedDerivations;
+  allGrammars = builtins.attrValues parsersWithMeta;
 
   # Usage:
   # pkgs.vimPlugins.nvim-treesitter.withPlugins (p: [ p.c p.java ... ])
   # or for all grammars:
   # pkgs.vimPlugins.nvim-treesitter.withAllGrammars
   withPlugins =
     f:
+    let
+      selectedGrammars = f (tree-sitter.builtGrammars // builtGrammars);
+
+      grammarPlugins = map grammarToPlugin selectedGrammars;
+
+      queryPlugins = lib.pipe selectedGrammars [
+        (map (g: g.passthru.associatedQuery or null))
+        (builtins.filter (q: q != null))
+      ];
+    in
     self.nvim-treesitter.overrideAttrs {
-      passthru.dependencies = map grammarToPlugin (f (tree-sitter.builtGrammars // builtGrammars));
+      passthru.dependencies = [
+        (symlinkJoin {
+          name = "nvim-treesitter-grammars";
+          paths = grammarPlugins ++ queryPlugins;
+        })
+      ];
     };
 
   withAllGrammars = withPlugins (_: allGrammars);
 in
 
 {
-  postPatch = ''
-    rm -r parser
-  '';
```

Comment:
Shouldn't we `rm -r runtime/queries`?

Reply:
No. I'm pretty sure it'll be possible for nix users to use nvim-treesitter to install parsers and queries after this PR.

`runtime/` is not on Neovim's runtimepath, so keeping the directory doesn't do any harm.